src/osd/DynamicPerfStats.h

```diff
@@ -0,0 +1,79 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef DYNAMIC_PERF_STATS_H
+#define DYNAMIC_PERF_STATS_H
+
+#include "mgr/OSDPerfMetric.h"
+#include "mgr/OSDPerfMetricReport.h"
+
+class DynamicPerfStats {
+public:
+  DynamicPerfStats() {
+  }
+
+  DynamicPerfStats(const std::list<OSDPerfMetricQuery> &queries) {
+    for (auto &query : queries) {
+      data[query];
+    }
+  }
+
+  void set_queries(const std::list<OSDPerfMetricQuery> &queries) {
+    std::map<OSDPerfMetricQuery, std::map<std::string, PerfCounters>> new_data;
+    for (auto &query : queries) {
+      new_data[query] = std::move(data[query]);
+    }
+    data = std::move(new_data);
+  }
+
+  bool is_enabled() {
+    return !data.empty();
+  }
+
+  void add(const MOSDOp *m, uint64_t inb, uint64_t outb, const utime_t &latency) {
+    for (auto &it : data) {
+      auto &query = it.first;
+      it.second[query(m)] += {1, inb, outb, latency.to_nsec()};
```

Comment:
Nit: need to differentiate between reads and writes. This can be addressed in a future PR.